src/cudamatrix/cu-device.h

```diff
@@ -55,7 +55,12 @@ class CuDevice {
   inline void* Malloc(size_t size) { return allocator_.Malloc(size); }
 
   inline void* MallocPitch(size_t row_bytes, size_t num_rows, size_t *pitch) {
-    return allocator_.MallocPitch(row_bytes, num_rows, pitch);
+    if (random_stride_mode_) {
+      return allocator_.MallocPitch(row_bytes + 128 * (rand() & 1), num_rows,
```

Comment:
Makes sense to me. I couldn't find it documented anywhere what the striding
requirements were for aligned accesses, but 128 seems high enough to get
into a different bucket anyway.

On Tue, Aug 23, 2016 at 7:42 PM, Daniel Povey <notifications@github.com>
wrote:

> In src/cudamatrix/cu-device.h
> <https://github.com/kaldi-asr/kaldi/pull/997#discussion_r75972231>:
>
> > @@ -55,7 +55,12 @@ class CuDevice {
> >    inline void* Malloc(size_t size) { return allocator_.Malloc(size); }
> >
> >    inline void* MallocPitch(size_t row_bytes, size_t num_rows, size_t *pitch) {
> > -    return allocator_.MallocPitch(row_bytes, num_rows, pitch);
> > +    if (random_stride_mode_) {
> > +      return allocator_.MallocPitch(row_bytes + 128 * (rand() & 1), num_rows,
>
> doesn't matter, as it will be rounded up to a multiple of 128 bytes anyway
> to get the stride. adding 128 puts it in a different bucket.
> … <#m_-1484969808801189986_>
> On Tue, Aug 23, 2016 at 7:31 PM, Daniel Galvez ***@***.***> wrote: In
> src/cudamatrix/cu-device.h <#997 (comment)
> <https://github.com/kaldi-asr/kaldi/pull/997#discussion_r75971055>>: > @@
> -55,7 +55,12 @@ class CuDevice { > inline void* Malloc(size_t size) {
> return allocator_.Malloc(size); } > > inline void* MallocPitch(size_t
> row_bytes, size_t num_rows, size_t *pitch) { > - return
> allocator_.MallocPitch(row_bytes, num_rows, pitch); > + if
> (random_stride_mode_) { > + return allocator_.MallocPitch(row_bytes + 128
> * (rand() & 1), num_rows, I actually question the usage of 128, as CUDA
> allocations tend to be aligned on some multiple of 2 (like 64) anyway. I
> would exepct that 129 would be more likely to expose buggy code. — You are
> receiving this because you modified the open/close state. Reply to this
> email directly, view it on GitHub <https://github.com/kaldi-asr/
> kaldi/pull/997/files/ccd2b85ea256cfe1e3f6ad76c04c4e260a0e6344#r75971055>,
> or mute the thread <https://github.com/notifications/unsubscribe-auth/
> ADJVu1afHQFqcRs38amOOonERkgNof-Gks5qi4LqgaJpZM4Jqu95> .
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/997/files/ccd2b85ea256cfe1e3f6ad76c04c4e260a0e6344#r75972231>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEi_UJLvW7aaYnbwdPiXFe2Ec0XGPVPOks5qi4VagaJpZM4Jqu95>
> .
>



-- 
Daniel Galvez